Added additional condition in getRegion() method to prevent Magento from throwing error in VatValidator::validate method.

Bug confirmed present in 2.3.0+. The fix should be officialy released with 2.3.3.

Original issue: https://github.com/magento/magento2/issues/22556
PR: https://github.com/magento/magento2/pull/22558


diff --git a/Model/Address/AbstractAddress.php b/Model/Address/AbstractAddress.php
index 5e42e42..e6bb0b4 100644
--- a/Model/Address/AbstractAddress.php
+++ b/Model/Address/AbstractAddress.php
@@ -375,6 +375,8 @@ class AbstractAddress extends AbstractExtensibleModel implements AddressModelInt
             }
         } elseif (is_string($region)) {
             $this->setData('region', $region);
+        }  elseif (!$regionId && is_array($region)) {
+            $this->setData('region', $regionId);
         }
 
         return $this->getData('region');
